Tools/clinic/libclinic/parse_args.py

```diff
@@ -518,30 +522,13 @@ def parse_pos_only(self) -> None:
         use_parser_code = True
         for i, p in enumerate(self.parameters):
             if p.is_vararg():
+                var = p.converter.parser_name
                 if self.fastcall:
-                    parser_code.append(libclinic.normalize_snippet("""
-                        %s = PyTuple_New(%s);
-                        if (!%s) {{
-                            goto exit;
-                        }}
-                        for (Py_ssize_t i = 0; i < %s; ++i) {{
-                            PyTuple_SET_ITEM(%s, i, Py_NewRef(args[%d + i]));
-                        }}
-                        """ % (
-                            p.converter.parser_name,
-                            left_args,
-                            p.converter.parser_name,
-                            left_args,
-                            p.converter.parser_name,
-                            self.max_pos
-                        ), indent=4))
+                    code = f"{var} = args + {self.vararg};"
                 else:
-                    parser_code.append(libclinic.normalize_snippet("""
-                        %s = PyTuple_GetSlice(%d, -1);
-                        """ % (
-                            p.converter.parser_name,
-                            self.max_pos
-                        ), indent=4))
+                    code = f"{var} = _PyTuple_CAST(args)->ob_item;"
```

Comment:
This is not compatible with limited_capi and does not work if there are other positional parameters.

Reply:
Are you addressing this with #122945?

Reply:
#122945 was created earlier.

This change creates conflicts with #122945. I'm trying to resolve them, but this is not easy.

Reply:
I know and I understand that; I was just wondering if you were going to address the shortcomings in this PR with #122945 (after resolving the conflicts), or if we should create a separate PR for addressing the shortcomings?

Reply:
Sorry, but I think that solving this separately will introduce conflict with https://github.com/python/cpython/pull/122945.